cpp/include/cudf/detail/null_mask.cuh

```diff
@@ -530,8 +530,7 @@ std::pair<rmm::device_buffer, size_type> segmented_null_mask_reduction(
   OffsetIterator first_bit_indices_end,
   OffsetIterator last_bit_indices_begin,
   null_policy null_handling,
-  bool initial_value_provided,
-  bool valid_initial_value,
+  std::optional<bool> valid_initial_value,
```

Comment:
BTW, the docstring above may need to be reformatted as it is too difficult to read.

Reply:
@ttnghia How should the docstring be reformatted?

Reply:
They should be formatted such that you can easily identify each new tag (@).
For example:
```
* @brief .....
*
* Long description.......
*
* @param input An input...........................................................
*              continuous line should be aligned with the starting position of the sentence above.
* ....
```

Reply:
There is no requirement to format docs this way. If it isn't automated or enforced by clang-format/clang-tidy, then it isn't a requirement.

Reply:
I tend to agree that whitespace formatting in docstrings is unenforceable unless it can be automated. I try to adhere to grammatical rules like avoiding periods if the parameter description is an incomplete sentence, but I wouldn't require reformatting here. I'm not even sure if I _like_ the offset alignment for line continuations.

Reply:
It is not required. However, it is helpful to have.

The documentation generated from doxygen is read by the users. However, the doxygen itself is read by internal developers. By having a clean, well-formatted doxygen you help other developers to read it easier. Paragraph alignment and empty lines help easier navigate and identify keywords and tags.

Reply:
How to align and how to break paragraphs is just personal preference. The main point here is to not write cluster paragraphs but instead write clean, easy to read and navigate documentations.

Reply:
Our team strategy is to minimize the amount of “personal preference” around code formatting and style, because it costs a _lot_ in developer time and review iterations. This is why we have clang-format and other pre-commit hooks. We sacrifice “personal preference” in order to streamline writing and reviewing. If this alignment cannot be enforced with an automatic tool via pre-commit, we should not leave PR review comments about alignment in docstrings.

Reply:
I hate to dig up this dead horse only to flog it again, but this result is actually even harder to read, after the suggested formatting change. The blocks all start at different spots because the parameter names all differ in length.
IMHO, IDEs tend to highlight the `@param param_name` well enough that parsing isn't difficult.
*IF* we're absolutely going to re-indent this, consider the following:
```c++
/**
 * @brief Create an output null mask whose validity is determined by the
 * validity of any/all elements of segments of an input null mask.
 *
 * @tparam OffsetIterator Random-access input iterator type.
 * @param bitmask Null mask residing in device memory whose segments will be
 *        reduced into a new mask.
 * @param first_bit_indices_begin Random-access input iterator to the beginning
 *        of a sequence of indices of the first bit in each segment (inclusive).
 * @param first_bit_indices_end Random-access input iterator to the end of a
 *        sequence of indices of the first bit in each segment (inclusive).
 * @param last_bit_indices_begin Random-access input iterator to the beginning
 *        of a sequence of indices of the last bit in each segment (exclusive).
 * @param null_handling If `null_policy::INCLUDE`, all elements in a segment
 *        must be valid for the reduced value to be valid. If `null_policy::EXCLUDE`,
 *        the reduction is valid if any element in the segment is valid.
 * @param stream CUDA stream used for device memory operations and kernel launches.
 * @param mr Device memory resource used to allocate the returned buffer's device memory.
 * @return A pair containing the reduced null mask and number of nulls.
 */
```
My vote is not to reformat.